.github/workflows/docbuild.yml

```diff
@@ -0,0 +1,99 @@
+name: Documentation Build
+
+on:
+  pull_request_target:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Run documentation build
+
+    steps:
+    - name: Checkout the code
+      uses: actions/checkout@v2
+      with:
+        path: ncs/nrf
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+
+    - name: cache-pip
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-doc-pip
+
+    - name: Install packages
+      run: |
+        sudo apt-get install -y ninja-build doxygen mscgen sshpass
+
+    - name: Install base dependencies
+      working-directory: ncs
+      run: |
+        pip3 install -U pip
+        pip3 install -U setuptools
+        export PATH="$HOME/.local/bin:$PATH"
+        pip3 install -r nrf/scripts/requirements-base.txt
+
+    - name: West init and update
+      working-directory: ncs
+      run: |
+        export PATH="$HOME/.local/bin:$PATH"
+        west init -l nrf
+        west update
+        west zephyr-export
+
+    - name: Install documentation dependencies
+      working-directory: ncs
+      run: |
+        pip3 install -r zephyr/scripts/requirements-doc.txt
+        pip3 install -r nrf/scripts/requirements-doc.txt
+
+    - name: Build documentation
+      working-directory: ncs
+      run: |
+        export PATH="$HOME/.local/bin:$PATH"
+        mkdir -p _build && cd _build
+        cmake -GNinja ../nrf/doc
+        ninja build-all
+
+    - name: Check build warnings
+      working-directory: ncs
+      run: |
+        cd _build
+        if [ -s Kconfig/sphinx.log ]; then
+          errors=$(cat Kconfig/sphinx.log)
+          echo "::error file=Kconfig/sphinx.log::$errors"
+          exit 1
+        fi
+        if [ -s mcuboot/sphinx.log ]; then
```

Comment:
Add a check for `zephyr/doc.warnings` like in https://github.com/zephyrproject-rtos/zephyr/blob/master/.github/workflows/doc-build.yml#L67

Reply:
Done.